lightning/restore/restore_test.go

```diff
@@ -26,30 +31,35 @@ func (s *restoreSuite) TestNewTableRestore(c *C) {
 	testCases := []struct {
 		name       string
 		createStmt string
-		errRegexp  string
 	}{
-		{"t1", "CREATE TABLE `t1` (`c1` varchar(5) NOT NULL)", ""},
-		{"t2", "CREATE TABLE `t2` (`c1` varchar(30000) NOT NULL)", "failed to ExecDDLSQL `mockdb`.`t2`:.*"},
-		{"t3", "CREATE TABLE `t3-a` (`c1-a` varchar(5) NOT NULL)", ""},
+		{"t1", "CREATE TABLE `t1` (`c1` varchar(5) NOT NULL)"},
+		// {"t2", "CREATE TABLE `t2` (`c1` varchar(30000) NOT NULL)"}, // no longer able to create this kind of table.
```

Comment:
is this case meaningless? or add some error cases?

Reply:
In this PR we no longer parses the `CREATE TABLE` DDL, and instead directly unmarshal the JSON result from TiDB (calling `tables.TableFromMeta`). So yeah this case becomes meaningless since either you can't create a `VARCHAR(30000)` in TiDB, or you can and produced a proper JSON which Lightning accepts without error.

Anyway, added a separate unit test to ensure malformed table info will produce an error.